frontend/simple/js/state.js

```diff
@@ -99,19 +117,45 @@ const getters = {
   unreadMessageCount (state) {
     let mailboxContract = store.getters.mailboxContract
     return mailboxContract && mailboxContract.messages.filter((msg) => !msg.read).length
+  },
+  // Logged In user's identity contract
+  identity (state) {
+    return state.users[state.loggedIn]
+  },
+  // Logged In user's picture
+  picture (state) {
+    let identity = store.getters.identity
+    return identity && state[identity] && state[identity].attributes.picture
+  },
+  // list of group names and contractIds
+  groupsByName (state) {
+    let groups = []
+    for (let [key, value] of Object.entries(state.contracts)) {
+      if (value === 'GroupContract') {
+        groups.push({groupName: state[key].groupName, contractId: key})
+      }
+    }
+    return groups
```

Comment:
Practice writing this in the functional way. See if you can use lodash to turn these 7 lines into 1 line.